lib/livebook_web/router.ex

```diff
@@ -108,6 +108,7 @@ defmodule LivebookWeb.Router do
       live "/sessions/:id/settings/runtime", SessionLive, :runtime_settings
       live "/sessions/:id/settings/file", SessionLive, :file_settings
       live "/sessions/:id/settings/app", SessionLive, :app_settings
+      live "/sessions/:id/settings/launch-app", SessionLive, :app_settings_and_launch
```

Comment:
I would rather have the same URL with a query parameter like `.../settings/app?launch=true`, in `SessionLive.handle_params` we can assign a boolean `app_deploy_on_save: params["launch"] == "true"` and we pass this down to the settings component :)

Reply:
@jonatanklosko my concern is that we would set this assign that would now exist "forever", even if app settings modal is no longer used.

Reply:
@josevalim it's the same with all handle_params assigns in general though, no?

We could nest assigns from params under `assigns.params` and it would be cleared on navigation. I can do that in a separate PR if you think it's worth it.

Reply:
Storing all of the assigns has other issues, like we keep a copy of all query parameters, which can keep unrelated thing. Unless we store as a subset in `assigns.params`?

However, I think we should provide the same sort of feature for Docker Deployment, we should automatically show the configuration instead of asking the user to click away and I don't think we want to insert a third action in such case.

Reply:
On the other hand, I have been thinking: wouldn't it be nice if we could allow the `@live_action` to be a tuple, so we could do:

      live "/sessions/:id/settings/launch-app", SessionLive, {:app_settings, :launch}
      live "/sessions/:id/settings/docker-app", SessionLive, {:app_settings, :docker}

But then I realized it is kind of the same as an atom anyway. So maybe the atom is fine. I am indifferent. It is your call. :) We can also leave this be and revisit when we change it for Docker.

Reply:
> Storing all of the parameters has other issues, like we keep a copy of all query parameters, which can keep unrelated thing. Unless we store as a subset in assigns.params?

Oh I didn't mean the params map, I meant all the assigns we currently set from params, we would put under some namespace in assigns, so it can be easily reset.

Not sure about Docker, but semantically it feels like we are on the same page, just from a different context, which params indicate. May be overthinking tho :)

Reply:
So let's ship this as is and revisit it when we align the other deployment flows. :)

Reply:
Ok :)

Reply:
Can I mark it as resolved? =)

Reply:
We can keep for the reference.

@josevalim feel free to merge whenever!

Reply:
@jonatanklosko would you want to change *all* handle_event?

Reply:
@jonatanklosko would you want to change *all* handle_params?

Reply:
All in SessionLive, since it's the long-lived LV, but could do in other LVs too if you think it's worth it.

Reply:
Sounds good. I will do the Docker one first, you can review and tackle the rest if you prefer. :) It just only 3 or 4 anyway.

Reply:
Sounds good :)